crates/client/data-availability/src/eigenda/mod.rs

```diff
@@ -0,0 +1,133 @@
+        // verifyBlob() is the primary function that needs to be invoked
+        //      This function will take the blobHeader and blobVerificationProof as inputs and
+        //      execute a series of checks to ensure the blob was signed for and stored properly in the EigenDA network.
+        Ok(())
+    }
+
+    async fn last_published_state(&self) -> Result<I256> {
+        Ok(I256::from(1))
+    }
+
+    fn get_da_metric_labels(&self) -> HashMap<String, String> {
+        [("name".into(), "eigenda".into())].iter().cloned().collect()
+    }
+
+    fn get_mode(&self) -> DaMode {
+        self.mode
+    }
+}
+
+impl EigenDaClient {
+    // EigenDA gRPC server(s) are not currently working with tonic
+    // instead we use a macro to fork a command to the command line to send the gRPC requests
+    async fn disperse_blob(&self, state_diff: Vec<U256>) -> Result<DisperseBlobResponse, std::io::Error> {
+        let payload = self.create_disperse_blob_payload(state_diff);
+        let output = grpcurl_command!(
+            "-proto", &self.config.proto_path,
+            "-d", &payload,
+            &self.config.grpc_provider,
+            "disperser.Disperser/DisperseBlob"
+        )?;
+        if output.status.success() {
+            let disperse_blob_response: DisperseBlobResponse = String::from_utf8(output.stdout).map_err(|err| {
+                std::io::Error::new(std::io::ErrorKind::Other, err.to_string())
+            })?.into();
+            return Ok(disperse_blob_response)
+        } else {
+            let error_message = String::from_utf8(output.stderr).map_err(|err| {
+                std::io::Error::new(std::io::ErrorKind::Other, err.to_string())
+            })?;
+            return Err(std::io::Error::new(std::io::ErrorKind::Other, error_message));
+        }
+    }
+
+    async fn get_blob_status(&self, request_id: String) -> Result<BlobStatusResponse, std::io::Error> {
+        let payload = Self::create_get_blob_status_payload(request_id);
+        let output = grpcurl_command!(
+            "-proto", &self.config.proto_path,
+            "-d", &payload,
+            &self.config.grpc_provider,
+            "disperser.Disperser/GetBlobStatus"
+        )?;
+        if output.status.success() {
+            let response: BlobStatusResponse = String::from_utf8(output.stdout).map_err(|err| {
+                std::io::Error::new(std::io::ErrorKind::Other, err.to_string())
+            })?.into();
+            return Ok(response);
+        } else {
+            let error_message = String::from_utf8(output.stderr).map_err(|err| {
+                std::io::Error::new(std::io::ErrorKind::Other, err.to_string())
+            })?;
+            return Err(std::io::Error::new(std::io::ErrorKind::Other, error_message))
+        }
+    }
+
+    fn create_disperse_blob_payload(&self, state_diff: Vec<U256>) -> String {
```

Comment:
Let's create a rust Struct and serialize that using serde, this format would be difficult to edit later on